napari/utils/_tests/test_notification_manager.py

```diff
@@ -155,3 +159,26 @@ def _raise():
     assert threading.excepthook == previous_threading_exhook
 
     assert all(isinstance(x, Notification) for x in store)
+
+
+def test_notification_manager_no_warning_duplication():
+    def fun():
+        warnings.showwarning(
+            UserWarning('This is a warning'),
+            category=UserWarning,
+            filename=__file__,
+            lineno=170,
```

Comment:
kind of a nit but I noticed that the lineno you specify here is somewhat variable. what do you think about always setting it to the line number where the expression begins? (166 in this case)

Reply:
Please add suggestions.